druid/src/widget/scroll.rs

```diff
@@ -456,7 +573,16 @@ impl<T: Data, W: Widget<T>> Widget<T> for Scroll<T, W> {
         self.child_size = size;
         self.child.set_layout_rect(ctx, data, env, size.to_rect());
         let self_size = bc.constrain(self.child_size);
-        let _ = self.scroll(Vec2::new(0.0, 0.0), self_size);
+
+        let scroll_to = ScrollTo::new(
+            Some(f64::max(self_size.width - size.width, self.scroll_offset.x)),
+            Some(f64::max(
+                self_size.height - size.height,
+                self.scroll_offset.y,
+            )),
+        );
+
+        ctx.submit_command(SCROLL_TO.with(scroll_to), ctx.widget_state.id);
```

Comment:
When you submit a command in `layout` or `lifecycle`, it won't be processed until (at least) the next frame. I think scrolling immediately is better.

Reply:
This was because the scroll callback takes an EventContext (so they can submit commands). I think I may have experimented with giving them the ContextState (and thats why I changed the visibility on that method)? Or something wrapped around it. Maybe need their own context that is minimal and just lets them do what we expect, that can be derived from any other context...